generated/typespec-ts/src/api/operations.ts

```diff
@@ -0,0 +1,1708 @@
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      body: {
+        training_file: job["trainingFile"],
+        validation_file: job["validationFile"],
+        model: job["model"],
+        hyperparameters: !job.hyperparameters
+          ? undefined
+          : { n_epochs: job.hyperparameters?.["nEpochs"] },
+        suffix: job["suffix"],
+      },
+    });
+}
+
+export async function _createFineTuningJobDeserialize(
+  result: CreateFineTuningJob200Response | CreateFineTuningJobDefaultResponse
+): Promise<FineTuningJob> {
+  if (isUnexpected(result)) {
+    throw result.body;
+  }
+
+  return {
+    id: result.body["id"],
+    object: result.body["object"],
+    createdAt: new Date(result.body["created_at"]),
+    finishedAt: new Date(result.body["finished_at"]),
+    model: result.body["model"],
+    fineTunedModel: result.body["fine_tuned_model"],
+    organizationId: result.body["organization_id"],
+    status: result.body["status"] as any,
+    hyperparameters: {
+      nEpochs: result.body.hyperparameters["n_epochs"] as any,
+    },
+    trainingFile: result.body["training_file"],
+    validationFile: result.body["validation_file"],
+    resultFiles: result.body["result_files"],
+    trainedTokens: result.body["trained_tokens"],
+    error:
+      result.body.error === null
+        ? null
+        : {
+            message: result.body.error["message"],
+            code: result.body.error["code"],
+            param: result.body.error["param"],
+          },
+  };
+}
+
+/**
+ * Creates a job that fine-tunes a specified model from a given dataset.
+ *
+ * Response includes details of the enqueued job including job status and the name of the
+ * fine-tuned models once complete.
+ *
+ * [Learn more about fine-tuning](/docs/guides/fine-tuning)
+ */
+export async function createFineTuningJob(
+  context: Client,
+  job: CreateFineTuningJobRequest,
+  options: CreateFineTuningJobOptions = { requestOptions: {} }
+): Promise<FineTuningJob> {
+  const result = await _createFineTuningJobSend(context, job, options);
+  return _createFineTuningJobDeserialize(result);
+}
+
+export function _listPaginatedFineTuningJobsSend(
+  context: Client,
+  options: ListPaginatedFineTuningJobsOptions = { requestOptions: {} }
+): StreamableMethod<
+  | ListPaginatedFineTuningJobs200Response
+  | ListPaginatedFineTuningJobsDefaultResponse
+> {
+  return context
+    .path("/fine_tuning/jobs")
+    .get({
+      ...operationOptionsToRequestParameters(options),
+      queryParameters: { after: options?.after, limit: options?.limit },
+    });
+}
+
+export async function _listPaginatedFineTuningJobsDeserialize(
+  result:
+    | ListPaginatedFineTuningJobs200Response
+    | ListPaginatedFineTuningJobsDefaultResponse
+): Promise<ListPaginatedFineTuningJobsResponse> {
+  if (isUnexpected(result)) {
+    throw result.body;
+  }
+
+  return {
+    object: result.body["object"],
+    data: (result.body["data"] ?? []).map((p) => ({
```

Comment:
[Compile Error]

src/api/operations.ts:411:5 - error TS2322: Type '{ id: string; object: "fine_tuning.job"; createdAt: Date; finishedAt: Date; model: string; fineTunedModel: string | null; organizationId: string; status: any; hyperparameters: { ...; }; ... 4 more ...; error: { ...; } | null; }[]' is not assignable to type 'FineTuningJob[]'.
  Type '{ id: string; object: "fine_tuning.job"; createdAt: Date; finishedAt: Date; model: string; fineTunedModel: string | null; organizationId: string; status: any; hyperparameters: { ...; }; ... 4 more ...; error: { ...; } | null; }' is not assignable to type 'FineTuningJob'.